website/docs/docs/troubleshooting/disabling-dangerous-contract-modification.md

```diff
@@ -0,0 +1,120 @@
+```
+
+If contract modification is disabled, the following error would be shown:
+
+```
+Cannot change the content of the pact for Example Consumer version 1.2.26 and provider Example Provider, as race conditions will cause unreliable results for can-i-deploy. Each pact must be published with a unique consumer version number. Some Pact libraries generate random data when a concrete value for a type matcher is not specified, and this can cause the contract to mutate - ensure you have given example values for all type matchers. For more information see https://docs.pact.io/go/versioning
+ {
+   "interactions": [
+     {
+-      "description": "a request to list TODOs"
++      "description": "a request to list TODOs 2"
+     }
+   ]
+ }
+```
+
+### Checking the logs (On-Premesis only)
+
+Check the start up logs of the PactFlow application to determine the value for the configuration property `allow_dangerous_contract_modification`. The logs will look like this (depending on your log format):
+
+```
+01:18:30 I -- allow_dangerous_contract_modification=false source=...
+```
+
+If the value is `false`, then no further action is required.
+
+## How to identify which applications have contracts which are being modified
+
+### SaaS
+
+Please open a [support case](https://support.smartbear.com/pactflow/message) and request the Customer Care team to provide a list of applications with modified contracts.
+
+### On-premesis
+
+Run the following SQL query against the Postgres database:
+
+```sql
+select distinct c.name as consumer_name
+from pact_publications pp
+join (select consumer_version_id
+    from pact_publications
+    group by consumer_version_id
+    having count(*) > 1 ) as mpp
+  on mpp.consumer_version_id = pp.consumer_version_id
+join pacticipants c
+  on pp.consumer_id = c.id
+order by 1
+```
+
+## How to avoid contract modification
+
+To ensure a pipeline is publishing contracts in the correct way, follow these guidelines:
+
+* Ensure each contract is published with a deterministic version number that *is* or *contains* the git sha, as per https://docs.pact.io/go/versioning
+* If the recommended approach to version numbers is already being used, then it is likely that random data is being used in a pact. You can identify what has changed between pacts by following [this documentation](https://docs.pactflow.io/docs/how_to#see-what-has-changed-in-a-pact). Random data in a pact is often cause by using matcher without providing an example value (in which case, the Pact library provides a random example value). Ensure all matchers have an example value provided. Otherwise, the random data might be being included directly in the test by using test data generation tools.
```

Comment:
Otherwise, the random data might be included directly in the test by using data generation tools.